Formula/bazel.rb

```diff
@@ -29,24 +30,43 @@ def install
     # Force Bazel ./compile.sh to put its temporary files in the buildpath
     ENV["BAZEL_WRKDIR"] = buildpath/"work"
     # Force Bazel to use openjdk@11
-    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec/"openjdk.jdk/Contents/Home"
     ENV["EXTRA_BAZEL_ARGS"] = "--host_javabase=@local_jdk//:jdk"
+    on_macos { ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec/"openjdk.jdk/Contents/Home" }
+    on_linux { ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec }
+    # Force Bazel to use Homebrew python
+    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
 
     (buildpath/"sources").install buildpath.children
 
     cd "sources" do
-      system "./compile.sh"
-      system "./output/bazel",
-             "--output_user_root",
-             buildpath/"output_user_root",
-             "build",
-             "scripts:bash_completion"
+      on_macos do
+        system "./compile.sh"
+        system "./output/bazel", "--output_user_root",
+                                 buildpath/"output_user_root",
+                                 "build",
+                                 "scripts:bash_completion"
+      end
+      on_linux do
+        # Bazel clears environment variables other than PATH during build, which
+        # breaks Homebrew shim scripts. We don't see this issue on macOS since
+        # the build uses a Bazel-specific wrapper for clang rather than the shim;
+        # specifically, it uses `external/local_config_cc/wrapped_clang`.
+        #
+        # The workaround here is to disable the Linux shim for C/C++ compilers.
+        # Remove this when a way to retain HOMEBREW_* variables is found.
+        with_env(CC: "/usr/bin/gcc", CXX: "/usr/bin/g++") do
```

Comment:
Note that the syntactically cleaner option of sharing macOS/Linux system commands and using ENV like:
```rb
ENV["CC"] = "/usr/bin/gcc"
ENV["CXX"] = "/usr/bin/g++"
```
fails `brew style` check. It is also possible that current workaround will fail a future style check.

---

Another workaround would be to add an unnecessary `gcc` dependency and use
```rb
ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].any_installed_version.major}"
ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{Formula["gcc"].any_installed_version.major}"
```
Though, based on Arch Linux, it seems GCC 11 may fail and we would need to use GCC 10 or patch the missing include.